### PR TITLE
Tune rules w/ more advanced changelog selectors

### PR DESCRIPTION
**Pull request by acunniffe**
As discussed in #116, some rules should only apply when parts of the spec are added, changed, or removed. There are other classes of rules, requirements, that must hold all the time (not matter the history), and when something is changed.

With these 5 selectors we can express everything we currently need. The issue with #116 is that we were using `requirement` in places where we should have been using `added` or `requirementOnChange` 

- This PR tunes the rules in accordance to the choices made with @cmars in a sync call this AM. 
- Uses a stable version of the libraries (demo PRs were on prerelease w/o thorough testing or stability guarantees) 
- We made `added` and `changed` API compatible with `requirement` -- so it's easy to swap selectors w/o having to rewrite your check's handlers, or reorder your function arguments. 

Re: point 3 -- if we need to further change these rules, the fix should be easy 👍 

Really happy with how this turned out and how many things we figured out in the process of triaging this since Friday. @cmars over to you. Snapshots are pretty hard to review here because so many of the rules have disabled themselves and only qualify themselves on change now.

### CODE REVIEW

File: src/rulesets/operations.ts [hunk 1]
```diff
@@ -47,6 +47,15 @@ export const rules = {
...
   },
+  operationIdSet: ({ operations }: SnykApiCheckDsl) => {
+    operations.requirement.must(
+      "have operationId",
+      (operation, context, docs) => {
+        docs.includeDocsLink(links.standards.operationIds);
+        expect(operation.operationId).to.be.ok;
+      },
+    );
+  },
```
Comment (acunniffe) [inline src/rulesets/operations.ts:58]: require an operationId, always
compliments rule that checks if it's correct on change

File: src/rulesets/operations.ts [hunk 2]
```diff
@@ -73,7 +82,7 @@ export const rules = {
     );
   },
   parameterCase: ({ operations }: SnykApiCheckDsl) => {
-    operations.requirement.must(
+    operations.requirementOnChange.must(
```
Comment (acunniffe) [inline src/rulesets/operations.ts:85]: checks if operationId is valid only when it is changed

File: src/rulesets/tests/operations.test.ts [hunk 3]
```diff
@@ -357,22 +357,5 @@ describe("operation parameters", () => {
...
-              name: "version",
-            },
-          ];
-          return spec;
-        })
-        .withRule(rules.parameterCase, emptyContext);
-
-      expect(result.results[0].passed).toBeTruthy();
-      expect(result).toMatchSnapshot();
-    });
```
Comment (acunniffe) [inline src/rulesets/tests/operations.test.ts:376]: this test was never testing what it said it was.

File: src/rulesets/tests/operations.test.ts [hunk 4]
```diff
@@ -46,7 +46,7 @@ describe("operationId", () => {
           delete spec.paths!["/example"]!.get!.operationId;
           return spec;
         })
-        .withRule(rules.operationId, emptyContext);
+        .withRule(rules.operationIdSet, emptyContext);
```
Comment (acunniffe) [inline src/rulesets/tests/operations.test.ts:50]: was running with the wrong rule

File: src/rulesets/operations.ts [hunk 5]
```diff
@@ -31,7 +31,7 @@ const preventParameterChange = (schemaProperty: string) => {
 
 export const rules = {
   operationId: ({ operations }: SnykApiCheckDsl) => {
-    operations.requirement.must(
+    operations.requirementOnChange.must(
       "have the correct operationId format",
```
Comment (acunniffe) [inline src/rulesets/operations.ts:35]: only check if operations have the right format on change

**Review by cmars — APPROVED** [review]

**Pull request merged**